DocumentsFromSnapshotMigration/src/test/java/com/rfs/ParallelDocumentMigrationsTest.java

```diff
@@ -304,7 +304,8 @@ private int migrateDocumentsSequentially(
         String targetAddress,
         AtomicInteger runCounter,
         Random clockJitter,
-        DocumentMigrationTestContext testContext
+        DocumentMigrationTestContext testContext,
+        ClusterVersion parserVersion
```

Comment:
Looks like this name got used in many places, this value is only parsed once after that it seems like its the source/target version number

Reply:
Tl;DR - Agreed, will switch

I went back/forth on this one.  The catch here is that the parser version can be different than the source version, and whether you care/know about the parser or source version is dependent on context.   As an example, we will probably end up using the ES 7.10 parsers for OS 1.3.  All that said, on further reflection, I think the bigger point here may be that knowing there is a difference between the parser version and source version is an abstraction leak.

Reply:
I think what you are calling out here is a bigger problem that we need to address.  There is the version information supplied by the user that might not be accurate vs the version of the snapshot artifacts or service responses.

I'm working on this in the metadata systems - since the scenario is a bit more interactive.  I'll iterate on this space in my upcoming PR and we can see how this aligns with what you are thinking.